soh/src/code/z_parameter.c

```diff
@@ -1413,15 +1413,18 @@ void Inventory_SwapAgeEquipment(void) {
         }
 
         // When becoming adult, remove swordless flag since we'll get master sword
+        // (Unless Master Sword is shuffled)
         // Only in rando to keep swordless link bugs in vanilla
-        if (gSaveContext.n64ddFlag) {
+        if (!gSaveContext.n64ddFlag || !Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD)) {
```

Comment:
Not sure I understand why this is being changed liked this, surely we want

`if (gSaveContext.n64ddFlag && !Randomizer_GetSettingValue(RSK_SHUFFLE_MASTER_SWORD))`

Reply:
What I wrote is along the lines of `if the file is not randomizer, or the randomizer setting for shuffling MS is OFF` (which can only be set in a rando file)
I wasn't too sure about how to write it in a positive case, but your suggestion would fail in vanilla since it's written as `if the file is randomizer and the randomizer setting for shuffling MS is OFF`. Unless having MS on the B button automatically clears the swordless flag?

Reply:
In vanilla I believe the aim is to retain the bugs it has regarding change age. The swordless flag is never changed when turning adult normally I believe, it is only removed when selecting a sword in the equipment menu

Reply:
Yeah, the original suggestion I had for "combining the if statements" was to look like what @inspectredc  posted above.

The other two sets of if statements were fine because we wanted to inverse them to avoid the `{;}`